ratpack-core/src/test/groovy/ratpack/http/client/HttpClientSmokeSpec.groovy

```diff
@@ -517,4 +519,73 @@ BAR
     text == "bar"
   }
 
+
+  def "can handle compressed responses"() {
+    given:
+    def stringBuf = new ByteArrayOutputStream()
+    def writer = new OutputStreamWriter(new GZIPOutputStream(stringBuf))
+    writer.write("bar")
+    writer.close()
+
+    otherApp {
+      get("foo") {
+        response.headers.add(CONTENT_ENCODING, "gzip")
```

Comment:
This is where it's being done https://github.com/ratpack/ratpack/blob/877b7a2b6e465944601712a92410525314c3a5da/ratpack-core/src/main/java/ratpack/server/internal/SmartHttpContentCompressor.java

I can't seem to find any tests specifically for this but there are some for file compression.

I do take your point though.  My concern is that I don't know what happens if the client sends the `Accept-Encoding` header in this case, will otherApp try to apply compression to an already compressed response??  I guess not or your tests wouldn't be passing.  Maybe we need to put other assertions in along the way to make sure the behaviour is as we expect.